opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -12,141 +12,129 @@
+    which contains the new value.
+
+    Args:
+        key: The key of the entry to set
+        value: The value of the entry to set
+        context: The context to copy, if None, the current context is used
+    """
+    if context is None:
+        context = get_current()
+    new_values = context.copy()
+    new_values[key] = value
+    return Context(new_values)
+
+
+def remove_value(
+    key: str, context: typing.Optional[Context] = None
+) -> Context:
+    """To remove a value, this method returns a new context with the key
+    cleared. Note that the removed value still remains present in the old
+    context.
+
+    Args:
+        key: The key of the entry to remove
+        context: The context to copy, if None, the current context is used
+    """
+    if context is None:
+        context = get_current()
+    new_values = context.copy()
+    new_values.pop(key, None)
+    return Context(new_values)
+
+
+def get_current() -> Context:
+    """To access the context associated with program execution,
+    the RuntimeContext API provides a function which takes no arguments
+    and returns a RuntimeContext.
+    """
+
+    global _RUNTIME_CONTEXT  # pylint: disable=global-statement
+    if _RUNTIME_CONTEXT is None:
+        # FIXME use a better implementation of a configuration manager to avoid having
+        # to get configuration values straight from environment variables
+
+        configured_context = environ.get(
+            "OPENTELEMETRY_CONTEXT", "default_context"
+        )  # type: str
+        try:
+            _RUNTIME_CONTEXT = next(
+                iter_entry_points("opentelemetry_context", configured_context)
+            ).load()()
+        except Exception:  # pylint: disable=broad-except
+            logger.error("Failed to load context: %s", configured_context)
+
+    return _RUNTIME_CONTEXT.get_current()  # type:ignore
+
+
+def set_current(context: Context) -> Context:
+    """To associate a context with program execution, the Context
+    API provides a function which takes a Context.
+
+    Args:
+        context: The context to use as current.
+    """
+    old_context = get_current()
+    _RUNTIME_CONTEXT.set_current(Context(context))  # type:ignore
+    return old_context
+
+
+def with_current_context(
+    func: typing.Callable[..., "object"]
+) -> typing.Callable[..., "object"]:
+    """Capture the current context and apply it to the provided func."""
+
+    caller_context = get_current()
+
+    def call_with_current_context(
+        *args: "object", **kwargs: "object"
+    ) -> "object":
+        try:
+            backup = get_current()
+            set_current(caller_context)
+            return func(*args, **kwargs)
+        finally:
+            set_current(backup)
+
+    return call_with_current_context
+
+
+__all__ = [
+    "get_value",
+    "set_value",
+    "remove_value",
+    "get_current",
+    "set_current",
+    "RuntimeContext",
+]
```

Comment:
I still think we're better off without `__all__`s: https://github.com/open-telemetry/opentelemetry-python/pull/213#discussion_r335248733.

Reply:
Removed